concent_api/concent_api/tests/test_check_middleman_port.py

```diff
@@ -0,0 +1,69 @@
+from django.test import override_settings
+from django.conf import settings
+from django.test import TestCase
+
+from concent_api.system_check import check_middleman_port
+from concent_api.system_check import create_error_45_middleman_port_is_not_set
+from concent_api.system_check import create_error_46_middleman_port_has_wrong_type
+from concent_api.system_check import create_error_47_middleman_port_has_wrong_value
+
+
+class TestMiddlemanPortCheck(TestCase):
+
+    @override_settings(
+        CONCENT_FEATURES=[
+            'middleman'
+        ]
+    )
+    def test_that_middleman_port_not_set_will_produce_error_when_middleman_is_in_concent_features(self):
+        del settings.MIDDLEMAN_PORT
+
+        errors = check_middleman_port()
+
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(errors[0], create_error_45_middleman_port_is_not_set())
+
+    @override_settings(
+        CONCENT_FEATURES=[]
+    )
+    def test_that_middleman_port_not_set_will_not_produce_error_when_middleman_is_not_in_concent_features(self):
+        del settings.MIDDLEMAN_PORT
+
+        errors = check_middleman_port()
+
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        CONCENT_FEATURES=[
+            'middleman'
+        ],
+        MIDDLEMAN_PORT='1'
+    )
+    def test_that_non_int_middleman_port_rate_setting_type_should_produce_error(self):
+        errors = check_middleman_port()
+
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(errors[0], create_error_46_middleman_port_has_wrong_type(settings.MIDDLEMAN_PORT))
+
+    @override_settings(
+        CONCENT_FEATURES=[
+            'middleman'
+        ],
+        MIDDLEMAN_PORT=100000
+    )
+    def test_that_out_of_range_middleman_port_setting_type_should_produce_error(self):
```

Comment:
But I think it would still require `django.test.TestCase` for `override_settings` to work correctly. So I think the only way to do this is to write more regular tests. Would that be ok ?

Reply:
you don't have to use `django.test.TestCase` to be able to use `override_settings`. `override_settings` can be used with `pytest` as well, but only in form of context manager.

Reply:
:+1: done